addons/mrp/mrp.py

```diff
@@ -783,9 +783,9 @@ def action_cancel(self, cr, uid, ids, context=None):
         # Put related procurements in exception
         proc_obj = self.pool.get("procurement.order")
         procs = proc_obj.search(cr, uid, [('production_id', 'in', ids)], context=context)
-        if procs:
-            proc_obj.message_post(cr, uid, procs, body=_('Manufacturing order cancelled.'), context=context)
-            proc_obj.write(cr, uid, procs, {'state': 'exception'}, context=context)
+        for proc in procs:
+            proc_obj.message_post(cr, uid, proc, body=_('Manufacturing order cancelled.'), context=context)
+            proc_obj.write(cr, uid, proc, {'state': 'exception'}, context=context)
```

Comment:
The error is with the `message_post` taking only one id, right?
The write can be set outside of the loop for better performance